test/e2e/web_console_plugin_test.go

```diff
@@ -0,0 +1,153 @@
+		Username("consoletest").
+		Email("consoletest@redhat.com").
+		EnsureMUR().
+		RequireConditions(ConditionSet(Default(), ApprovedAutomatically())...).
+		Execute(s.T())
+	response := signupRequest.GetSignupResponse(s.T())
+
+	_, mur := signupRequest.Resources()
+
+	memberAwait := GetMurTargetMember(s.T(), s.Awaitilities, mur)
+
+	// Response should contain a ConsoleURL with a value something like:
+	// https://console-openshift-console.apps.99b682869228f7464338-mgmt.ci.hypershift.devcluster.openshift.com/
+	require.Contains(s.T(), response, "consoleURL")
+
+	// Since we can't easily access the web console API resources directly (due to complex security requirements) we
+	// will instead create a route in the member cluster with which to access the console plugin
+
+	// Before we can create the route however, we need the certificate and private key values from the
+	// member-operator-console-plugin secret
+	secret, err := memberAwait.WaitForSecret(s.T(), memberAwait.Namespace, "member-operator-console-plugin")
+	require.NoError(s.T(), err)
+
+	// Now we can create the route resource
+	route := &v1.Route{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "consolepluginroute",
+			Namespace: memberAwait.Namespace,
+			Annotations: map[string]string{
+				"openshift.io/host.generated": "true",
+			},
+		},
+		Spec: v1.RouteSpec{
+			To: v1.RouteTargetReference{
+				Kind: "Service",
+				Name: "member-operator-console-plugin",
+			},
+			Port: &v1.RoutePort{
+				TargetPort: intstr.FromInt(9443),
+			},
+			TLS: &v1.TLSConfig{
+				Termination: "reencrypt",
+				Certificate: string(secret.Data["tls.crt"]),
+				Key:         string(secret.Data["tls.key"]),
+			},
+			WildcardPolicy: "None",
+		},
+	}
+
+	err = memberAwait.Create(s.T(), route)
+	require.NoError(s.T(), err)
+
+	reloaded, err := memberAwait.WaitForRouteToBeAvailable(s.T(), route.Namespace, route.Name, "/status")
+	require.NoError(s.T(), err, "route not available", route)
+
+	// Construct the routeURL from the route host
+	routeURL := reloaded.Spec.Host + "/"
+
+	manifestURL := fmt.Sprintf("%s%s%s", "https://", routeURL, "plugin-manifest.json")
+	healthCheckURL := fmt.Sprintf("%s%s%s", "https://", routeURL, "status")
+
+	var tr *http.Transport
+	{ // #nosec G402
+		tr = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	httpClient := &http.Client{Transport: tr}
+
+	// First perform a health check - if the health check doesn't pass within one minute then the test will fail
+	var healthCheckResponse *http.Response
+
+	req, err := http.NewRequest("GET", healthCheckURL, nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", signupRequest.GetToken())
+
+	err = wait2.Poll(time.Second*5, time.Minute, func() (bool, error) {
+		healthCheckResponse, err = httpClient.Do(req) //nolint
+		if err != nil {
+			return false, err
+		}
+		defer healthCheckResponse.Body.Close()
+		return true, nil
+	})
```

Comment:
what is the point of running the `Poll` if the only values that you return are:
* `false, err` (which means that it immediately ends because of the error)
* `true, nil` (which means that it immediately ends because you return true)

in other words, there will be only one execution of the logic